plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -221,6 +228,14 @@ public IcebergMetadata(
         this.commitTaskCodec = requireNonNull(commitTaskCodec, "commitTaskCodec is null");
         this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableOperationsProvider is null");
         this.trinoVersion = requireNonNull(trinoVersion, "trinoVersion is null");
+        this.tableCache = CacheBuilder.newBuilder()
+                .maximumSize(100)
+                .expireAfterWrite(5, TimeUnit.MINUTES)
+                .build(new TableCacheLoader());
+        this.tableStatisticsCache = CacheBuilder.newBuilder()
```

Comment:
if adding stats cache is an improvement, i would expect this to be reflected by a test change.
Currently i see `TestIcebergMetadataFileOperations` passes without modification.

Let's move statistics to separate PR (or at least separate commit), so that we can focus on metastore interactions here.